polaris/polaris/management/commands/poll_pending_deposits.py

```diff
@@ -83,6 +87,90 @@ def check_for_multisig(transaction):
+    ``Transaction`` object passed. It also must not be used to submit transactions by any service
+    other than Polaris. If it is, the outstanding transactions will be invalidated due to bad
+    sequence numbers. Finally, the channel accounts must have a master signer with a weight greater
+    than or equal to the medium threshold for the account.
+
+    After the transaction for creating the destination account has been submitted to the stellar network
+    and the transaction has been created, this function will return the account and True. If the
+    transaction was not submitted successfully, a RuntimeError exception will be raised.
+    """
+    try:
+        account, json_resp = get_account_obj(
+            Keypair.from_public_key(transaction.stellar_account)
+        )
+        return account, False, has_trustline(transaction, json_resp)
+    except RuntimeError:
+        master_signer = None
+        if transaction.asset.distribution_account_master_signer:
+            master_signer = transaction.asset.distribution_account_master_signer
+        thresholds = transaction.asset.distribution_account_thresholds
+        if master_signer and master_signer["weight"] >= thresholds["med_threshold"]:
+            source_account_kp = Keypair.from_secret(transaction.asset.distribution_seed)
+            source_account, _ = get_account_obj(source_account_kp)
+        else:
+            from polaris.integrations import registered_deposit_integration as rdi
+
+            rdi.create_channel_account(transaction)
+            source_account_kp = Keypair.from_secret(transaction.channel_seed)
+            source_account, _ = get_account_obj(source_account_kp)
+
+        base_fee = settings.HORIZON_SERVER.fetch_base_fee()
+        builder = TransactionBuilder(
+            source_account=source_account,
+            network_passphrase=settings.STELLAR_NETWORK_PASSPHRASE,
+            base_fee=base_fee,
+        )
+        transaction_envelope = builder.append_create_account_op(
+            destination=transaction.stellar_account,
+            starting_balance=settings.ACCOUNT_STARTING_BALANCE,
+        ).build()
+        transaction_envelope.sign(source_account_kp)
+
+        try:
+            settings.HORIZON_SERVER.submit_transaction(transaction_envelope)
+        except BaseHorizonError as submit_exc:  # pragma: no cover
+            raise RuntimeError(
+                "Horizon error when submitting create account to horizon: "
+                f"{submit_exc.message}"
+            )
+
+        transaction.status = Transaction.STATUS.pending_trust
+        transaction.save()
+        logger.info(
+            f"Transaction {transaction.id} is now pending_trust of destination account"
+        )
+        account, json_resp = get_account_obj(
+            Keypair.from_public_key(transaction.stellar_account)
+        )
+        return account, True, has_trustline(transaction, json_resp)
```

Comment:
No need to check the JSON response of an account that was just created, it definitely doesn't have a trustline.
```suggestion
        account, _ = get_account_obj(
            Keypair.from_public_key(transaction.stellar_account)
        )
        return account, True, False
```

Reply:
Actually I believe this need to be `True`
Unless I'm misreading this `has_trustlines` returns false if there is a trustline and true if there isnt
This is probable because of the lose association with the `pending_trust` status
```
def has_trustline(transaction, json_resp):
    pending_trust = True
    for balance in json_resp["balances"]:
        if balance.get("asset_type") == "native":
            continue
        asset_code = balance["asset_code"]
        asset_issuer = balance["asset_issuer"]
        if (
            transaction.asset.code == asset_code
            and transaction.asset.issuer == asset_issuer
        ):
            pending_trust = False
            break
    return pending_trust
```